src/components/button.js

```diff
@@ -0,0 +1,121 @@
+    },
+    shaded: {
+      default: { text: 'text-primary-700', hover: 'hover:border-primary-700' },
+      normal: { text: 'text-success', hover: 'hover:border-success' },
+      warning: { text: 'text-warning', hover: 'hover:border-warning' },
+      error: { text: 'text-error', hover: 'hover:border-error' },
+      textDisabled: 'text-secondary-500',
+    },
+    filled: {
+      default: { bgColor: 'bg-primary-700', hover: 'hover:bg-primary-800' },
+      normal: { bgColor: 'bg-success', hover: 'hover:bg-success-hover' },
+      warning: { bgColor: 'bg-warning', hover: 'hover:bg-warning-hover' },
+      error: { bgColor: 'bg-error', hover: 'hover:bg-error-hover' },
+      bgColorDisabled: 'bg-secondary-500',
+    },
+  },
+  sizes: {
+    lg: 'h-9 py-2 px-4',
+    md: 'h-6.5 py-1 px-2.5',
+    sm: 'h-5 py-px px-1',
+    squared: {
+      lg: 'p-2.5',
+      md: 'p-1.5',
+      sm: 'p-1',
+    },
+    text: {
+      lg: 'text-sm tracking-sm',
+      md: 'text-xs tracking-md',
+      sm: 'text-xxs tracking-lg',
+    },
+    iconPadding: {
+      startIcon: {
+        sm: 'pr-2.5',
+        md: 'pr-3',
+        lg: 'pr-3.5',
+      },
+      endIcon: {
+        sm: 'pl-2.5',
+        md: 'pl-3',
+        lg: 'pl-3.5',
+      },
+    },
+  },
+})
+
+const Button = ({ children, variant, size, color, disabled, ...rest }) => {
+  const { colors, sizes } = buttonConfigs()
+  const variants = {
+    outlined: clsx('border border-1', {
+      [`${colors[variant].borderDisabled} cursor-default ${colors[variant].textDisabled}`]: disabled,
+      [`${colors[variant][color].border} ${colors[variant][color].hover} ${colors[variant][color].text}`]: !disabled,
+    }),
+    borderless: clsx({
+      [`${colors[variant].bgColorDisabled} cursor-default ${colors[variant].textDisabled}`]: disabled,
+      [`${colors[variant][color].bgColor} ${colors[variant][color].hover} ${colors[variant][color].text}`]: !disabled,
+    }),
+    shaded: clsx('border border-1 border-white bg-white', {
+      [`${colors[variant].textDisabled} cursor-default shadow-10`]: disabled,
+      [`${colors[variant][color].text} ${colors[variant][color].hover} shadow-blue-10 hover:shadow-blue-30`]: !disabled,
+    }),
+    filled: clsx('text-white', {
+      [`${colors[variant].bgColorDisabled} cursor-default`]: disabled,
+      [`${colors[variant][color].bgColor} ${colors[variant][color].hover}`]: !disabled,
+    }),
+  }
+
+  const classes = {
+    button: clsx(`focus:outline-none rounded-sm ${sizes.text[size]} font-body font-normal ${variants[variant]}`, {
+      [sizes[size]]: typeof children === 'string',
+      [sizes.squared[size]]: typeof children !== 'string',
+    }),
+    startIcon: sizes.iconPadding.startIcon[size],
+    endIcon: sizes.iconPadding.endIcon[size],
+  }
+
+  return (
+    <ButtonBase classes={classes} disabled={disabled} {...rest}>
+      {(typeof children === 'string' && size === 'sm') ? children.toUpperCase() : children}
```

Comment:
not sure about performance/memory consumption between the CSS text transformation vs. JS string transformation.

however, take into consideration of misuse of this component, where `children` can be anything, _not_ always a string (and naturally so as a good children-wrapper component), such as the following case shouldn't be constrained:

```jsx
<Button>
    <span>lol</span>
</Button>
```

in such case, CSS text transformation becomes a bit more appealing as a more robust solution:

```jsx
const ButtonBase = (...) => {
  <button className={{..., size determination logic to apply tailwind's .uppercase class or not }} ...>
    ...
  </button>
}
```

may not be the best place, perhaps simpler to wrap the `children` inside the `Button` component with a transient layer of `<span>` that allows you to apply this class:

```jsx
const Button = (...) => {
  ...
  return (
    <ButtonBase ...>
      <span className={size === 'sm' ? 'uppercase' : ''}>
        {children}
      </span>
    </ButtonBase>
  )
} 
```

or wrap the condition outside of `<span>` to render `children` directly if the `size` isn't `sm`

Reply:
ohh oki, will change to css text-transformation